fhir-server/src/main/java/au/csiro/pathling/fhirpath/literal/CodingLiteralPath.java

```diff
@@ -50,29 +51,54 @@ protected CodingLiteralPath(@Nonnull final Dataset<Row> dataset, @Nonnull final
    * @return A new instance of {@link LiteralPath}
    * @throws IllegalArgumentException if the literal is malformed
    */
+  @Nonnull
   public static CodingLiteralPath fromString(@Nonnull final String fhirPath,
       @Nonnull final FhirPath context) throws IllegalArgumentException {
     final LinkedList<String> codingTokens = new LinkedList<>(Arrays.asList(fhirPath.split("\\|")));
     final Coding coding;
     if (codingTokens.size() == 2) {
-      coding = new Coding(codingTokens.get(0), codingTokens.get(1), null);
+      coding = new Coding(decodeComponent(codingTokens.get(0)),
+          decodeComponent(codingTokens.get(1)), null);
     } else if (codingTokens.size() == 3) {
-      coding = new Coding(codingTokens.get(0), codingTokens.get(2), null);
-      coding.setVersion(codingTokens.get(1));
+      coding = new Coding(decodeComponent(codingTokens.get(0)),
+          decodeComponent(codingTokens.get(2)), null);
+      coding.setVersion(decodeComponent(codingTokens.get(1)));
     } else {
       throw new IllegalArgumentException(
           "Coding literal must be of form [system]|[code] or [system]|[version]|[code]");
     }
     return new CodingLiteralPath(context.getDataset(), context.getIdColumn(), coding);
   }
 
+  @Nonnull
+  private static String decodeComponent(@Nonnull final String component) {
+    if (component.matches("^'(.*)'$")) {
+      final String result = unSingleQuote(component);
+      return StringLiteralPath.unescapeFhirPathString(result);
+    } else {
+      return component;
+    }
+  }
+
+  @Nonnull
+  private static String encodeComponent(@Nonnull final String component) {
+    if (component.matches("[',]")) {
```

Comment:
This looks suspicious: 
Why ',' should indicate need for escaping.
Also probably should be component.matches(".*[',].*")

Reply:
I've changed this to match the same pattern that is used for `RAW_CODING_COMPONENT`, which is any character except those used to delimit strings, arguments and coding components.